menuconfig.py

```diff
@@ -165,100 +165,255 @@
+                        "Error near: {}".format(t))
 
-    # Body of dialog boxes
-    DIALOG_BODY_STYLE        = (curses.COLOR_WHITE, curses.COLOR_BLACK,  curses.A_NORMAL                   )
+            cnum = int(cnum)
+            if not -1 <= cnum <= curses.COLORS:
+                raise StyleError(
+                        "Fixed color must be in range 0..{}. " \
+                        "Error near: {}".format(curses.COLORS - 1, t))
 
-    # Edit box in pop-up dialogs
-    DIALOG_EDIT_STYLE        = (curses.COLOR_WHITE, curses.COLOR_BLUE,   curses.A_NORMAL, curses.A_STANDOUT)
+            return int(cnum)
 
-    # Edit box in jump-to dialog
-    JUMP_TO_EDIT_STYLE       = (curses.COLOR_WHITE, curses.COLOR_BLUE,   curses.A_NORMAL,                  )
+        else:
+            raise StyleError("Unknown color definition: {}".format(t))
+
+    attrs = 0
+    fg_color = -1
+    bg_color = -1
+
+    # Parse attributes
+    for t in cstr.split(","):
+        if t == "": continue
+
+        if t == "bold":
+            # A_BOLD tends to produce faint and hard-to-read text on the Windows
+            # console, especially with the old color scheme, before the
+            # introduction of
+            # https://blogs.msdn.microsoft.com/commandline/2017/08/02/updating-the-windows-console-colors/
+            attrs |= curses.A_NORMAL if _IS_WINDOWS else curses.A_BOLD
+        elif t == "standout":
+            attrs |= curses.A_STANDOUT
+        elif t == "underline":
+            attrs |= curses.A_UNDERLINE
+        elif t.startswith("fg:"):
+            fg_color = parse_color(t)
+        elif t.startswith("bg:"):
+            bg_color = parse_color(t)
+        else:
+            raise StyleError("Invalid style attribute: {}".format(cstr))
 
-    # Symbol information text
-    INFO_TEXT_STYLE          = LIST_STYLE
+    return _style(fg_color, bg_color, attrs)
 
-    if os.environ.get("MENUCONFIG_THEME") == "aquatic":
-        # More 'make menuconfig'-like theme, contributed by Mitja Horvat
-        # (pinkfluid)
-        PATH_STYLE         = (curses.COLOR_CYAN,  curses.COLOR_BLUE,  BOLD                              )
-        SEPARATOR_STYLE    = (curses.COLOR_WHITE, curses.COLOR_CYAN,  BOLD,            curses.A_STANDOUT)
-        HELP_STYLE         = PATH_STYLE
-        DIALOG_FRAME_STYLE = SEPARATOR_STYLE
-        DIALOG_BODY_STYLE  = (curses.COLOR_WHITE, curses.COLOR_BLUE,  curses.A_NORMAL                   )
-        DIALOG_EDIT_STYLE  = (curses.COLOR_BLACK, curses.COLOR_WHITE, curses.A_NORMAL, curses.A_STANDOUT)
+def _init_styles():
+    global _PATH_STYLE
+    global _SEPARATOR_STYLE
+    global _LIST_STYLE
+    global _LIST_SEL_STYLE
+    global _LIST_INVISIBLE_STYLE
+    global _LIST_INVISIBLE_SEL_STYLE
+    global _HELP_STYLE
+    global _DIALOG_FRAME_STYLE
+    global _DIALOG_BODY_STYLE
+    global _DIALOG_EDIT_STYLE
+    global _JUMP_TO_EDIT_STYLE
+    global _INFO_TEXT_STYLE
```

Comment:
Guess these "bounce" variables are a bit silly now. Could just have a global `_style` dictionary instead, and reference that directly in the code.

I could fix that myselt though. Bit sadistic to force you to change a ton of unrelated code.

Reply:
Shouldn't be too hard, I'll give it a shot.

Reply:
Replaced all _*_STYLE variables with a single global _style dictionary.

Reply:
Thanks for fixing it for me!

Will take a look at the rest tomorrow.

Reply:
One thing you could do in the meantime if you want:

The final commit message should just describe the final change. No need to mention the fixes from earlier versions.

Reply:
Might look tomorrow instead. Pretty busy day today. :/